homeassistant/components/vacuum/roomba.py

```diff
@@ -0,0 +1,301 @@
+"""
+Support for Wi-Fi enabled iRobot Roombas.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.roomba/
+"""
+from functools import partial
+import asyncio
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.vacuum import (
+    VacuumDevice,
+    PLATFORM_SCHEMA, SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_PAUSE,
+    SUPPORT_STOP, SUPPORT_RETURN_HOME, SUPPORT_BATTERY, SUPPORT_STATUS,
+    SUPPORT_SEND_COMMAND)
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, CONF_NAME, CONF_HOST, CONF_USERNAME, CONF_PASSWORD)
+
+REQUIREMENTS = ['https://github.com/pschmitt/Roomba980-Python/archive/'
```

Comment:
I already opened an issue regarding this here: https://github.com/NickWaterton/Roomba980-Python/issues/3

But yeah I'll probably end up uploading it myself.